// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "google/protobuf/empty.proto";

package users.v1;

message User {
  string first_name = 1;
  string last_name = 2;
  uint64 id = 3;
}

message AddUserResponse {
  uint64 user_id = 1;
}

service Users {
  rpc AddUser(User) returns (AddUserResponse) {}
  rpc GetUsers(google.protobuf.Empty) returns (stream User) {}
}
